packages/db/tests/test-providers.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable unicorn/no-null */
+const test = require('ava');
+
+const {
+  translateApiProviderToPostgresProvider,
+  nullifyUndefinedProviderValues,
+  encryptValueWithKMS,
+} = require('../dist/provider');
+
+test('translateApiProviderToPostgresProvider translates a Cumulus Provider object to a Postgres Provider object', async (t) => {
+  const fakeEncryptFunction = async () => 'fakeEncryptedString';
+  const cumulusProviderObject = {
+    id: 'testId',
+    globalConnectionLimit: 1,
+    protocol: 'fakeProtocol',
+    host: 'fakeHost',
+    port: 1234,
+    username: 'fakeUsername',
+    password: 'fakePassword',
+    encrypted: true,
+    createdAt: 1234,
+    updatedAt: 5678,
+    privateKey: 'fakeKey',
+    cmKeyId: 'fakecmId',
+    certificateUri: 'fakeUri',
+  };
+
+  const expected = {
+    certificateUri: 'fakeUri',
+    cmKeyId: 'fakecmId',
+    created_at: 1234,
+    globalConnectionLimit: 1,
+    host: 'fakeHost',
+    name: 'testId',
+    password: 'fakeEncryptedString',
+    port: 1234,
+    privateKey: 'fakeKey',
+    protocol: 'fakeProtocol',
+    updated_at: 5678,
+    username: 'fakeEncryptedString',
+  };
+  const result = await translateApiProviderToPostgresProvider(
+    cumulusProviderObject,
+    fakeEncryptFunction
+  );
+  t.deepEqual(result, expected);
+});
+
+test('nullifyUndefinedProviderValues sets undefined provider values to "null"', async (t) => {
+  const cumulusProviderObject = {
+    name: 'fakeName',
+    protocol: 'fakeProtocol',
+    host: 'fakeHost',
+    port: 'fakePort',
+  };
+
+  const expected = {
+    name: 'fakeName',
+    protocol: 'fakeProtocol',
+    host: 'fakeHost',
+    port: 'fakePort',
+    username: null,
+    password: null,
+    globalConnectionLimit: null,
+    privateKey: null,
+    cmKeyId: null,
+    certificateUri: null,
+  };
+
+  const actual = nullifyUndefinedProviderValues(cumulusProviderObject);
+  t.deepEqual(actual, expected);
+});
+
+test('encryptValueWithKMS throws if provder_kms_key_id is undefined', async (t) => {
+  t.throws(() => encryptValueWithKMS('somevalue'));
+});
+
+test.serial('encryptValueWithKMS encrypts the key', async (t) => {
+  const actual = await encryptValueWithKMS('somevalue', () => 'encrypted', 'kmsKey');
```

Comment:
Seems like we're testing two things here: that a custom encrypt function is used and that when you provide a  custom KMS key, no error is thrown. I think it might be better to specify an env var for this test and separate the test for passing a custom KMS key ID.

Reply:
If we're going that route, I'll just pull out the custom ID logic and use a mock, as it serves no other purpose than to avoid a perhaps less straightforward proxyquire setup.

Reply:
Well, I wouldn't go that far, personally. I think having a function that is easy to test is all for the best.

I don't understand the reason not to set the env var for this test, since it's not used with a custom encrypt function anyway.

And I'm fine with keeping the custom KMS key parameter, though are we really using it/relying on it?

Reply:
I mostly have a mild dislike for using env vars in this way, but I suppose a one-off that resets it is fine in this case. 

How about: fc8e4394f